katas/README.md

```diff
@@ -0,0 +1,37 @@
+# Katas
+
+A Kata is a top-level container of educational items which are used to explain a particular quantum computing topic using Q#. They are organized in sections which can be of three types:
+- Lessons: text content that sometimes contains Q# code examples.
+- Exercises: problems that the user solves by writting Q# code.
+- Questions: analytical problems that have a text answer.
+
+## Rust crate
+
+The katas crate exposes an API to check solutions for exercises.
+
+## Composition
+
+Katas are composed through an index.md markdown file that uses macros to indicate how to produce content for different kinds of sections.
+
+### Macros
+
+The following macros are available for katas composition:
+- @[exercise]: Used to create Q# code exercises that we can be automatically verified.
```

Comment:
Your regular expression for finding macros requires that after the macro name (e.g. `@[exercise]`) that it be following by `({...})\n` (where ... can be any content over more than one line). You might want to outline that requirement here (any maybe some examples).

Reply:
Updated the readme to provide a better specification of macros.

I didn't include examples since we will probably revisit the authoring of katas soon and some macros might not be needed anymore.